internal/pkg/service/stream/keboolasink/bridge/file.go

```diff
@@ -115,62 +116,67 @@ func (b *Bridge) deleteCredentialsOnFileDelete() {
 	})
 }
 
-func (b *Bridge) registerFileImporter() {
-	b.plugins.RegisterFileImporter(
-		targetProvider,
-		func(ctx context.Context, file *plugin.File) error {
-			// Get authorization token
-			var token *keboolasink.Token
-			err := b.schema.Token().ForSink(file.SinkKey).GetOrNil(b.client).WithResultTo(&token).Do(ctx).Err()
-			if err != nil {
-				return err
-			}
+func (b *Bridge) importFile(ctx context.Context, file *plugin.File, stats statistics.Value) error {
+	start := time.Now()
 
-			// Get file upload credentials
-			var keboolaFile keboolasink.File
-			err = b.schema.File().ForFile(file.FileKey).GetOrErr(b.client).WithResultTo(&keboolaFile).Do(ctx).Err()
-			if err != nil {
-				return err
-			}
+	// Get authorization token
+	token, err := b.schema.Token().ForSink(file.SinkKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
+	if err != nil {
+		return err
+	}
 
-			// Authorized API
-			api := b.publicAPI.WithToken(token.TokenString())
+	// Get file details
+	keboolaFile, err := b.schema.File().ForFile(file.FileKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
+	if err != nil {
+		return err
+	}
 
-			// Check if job already exists
-			var job *keboola.StorageJob
-			if keboolaFile.StorageJobID != nil {
-				job, err = api.GetStorageJobRequest(keboola.StorageJobKey{ID: *keboolaFile.StorageJobID}).Send(ctx)
-				if err != nil {
-					return err
-				}
+	// Authorized API
+	api := b.publicAPI.WithToken(token.TokenString())
 
-				if job.Status == keboola.StorageJobStatusSuccess {
-					return nil
-				}
-			}
+	defer func() {
+		ctx, cancel := context.WithTimeout(ctx, uploadEventSendTimeout)
```

Comment:
I'll build a branch above this to set `uploadEventSendTimeout` configurable.

Reply:
Just add comment here  https://github.com/keboola/keboola-as-code/pull/1968/files#diff-61f7272294f4d27b17650f7b104d5c118a303c08bb484f0c9f656bc600b0a868R10 :slightly_smiling_face:

Reply:
already adjusted it. Probablly won't be reconfigured.